lib/exercises.rb

```diff
@@ -1,23 +1,51 @@
-
-
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
-
+# Time Complexity: O(n)
+# Space Complexity: O(n^2)
+strings = ["eat", "tea", "tan", "ate", "nat", "bat"]
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  hash = {}
+  # seperate words, alphabetizes them, and puts them in a hash
+  strings.each do |word|
+    letters = word.split('').sort().join()
+    hash[letters] = [] if hash[letters].nil?
+    hash[letters] << word
+  end
+  
+  # sorts through has by the key and pushes to array
+  anagrams = []
+  hash.keys.each do |key|
+    anagrams << hash[key]
+  end
+  return anagrams
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n^2)
 def top_k_frequent_elements(list, k)
```

Comment:
The space complexity is O(n) and the time is O(nk) see below.  Otherwise this works.  Let me know what questions you have.